test/signup.spec.js

```diff
@@ -0,0 +1,60 @@
+const supertest = require('supertest');
+
+const app = require('../app');
+const {
+  mockUser,
+  mockUserInvalidPass,
+  mockUserMissingFirstName,
+  mockUserMissingLastName,
+  mockUserMissingPassword,
+  mockUserMissingEmail
+} = require('./mocks/mockUsers.js');
+
+const request = supertest(app);
+const createUser = user => request.post('/users').send(user);
+describe('POST /users', () => {
+  it('should respond succesfully if it passes all validations', () =>
+    createUser(mockUser).then(res => {
+      expect(res.statusCode).toBe(201);
+      expect(res.body).toHaveProperty('userId');
+    }));
+
+  it('should fail with error if email already exist', () =>
+    createUser(mockUser).then(res => {
+      expect(res.statusCode).toBe(503);
+      expect(res.body.internal_code).toContain('database_error');
```

Comment:
You can use the const for `database_error` (stored in errors middleware if I remember right)

Reply:
change it to the error constant